### PR TITLE
Fixed Angle.angle when angle~=180.

### LINKED ISSUE #1556: angle.angle() returns nan when angle is 180
### Expected behaviour
angle.angle() to return the angle in degrees.

### Actual behaviour
angle.angle() returns nan when angle is 180 degrees.

### Note
This seems to happen because float rounding can make the angle bigger than 180.

### Code to reproduce the behaviour

``` python
In [1]:import MDAnalysis as mda
       from MDAnalysis.core.topologyattrs import Angles
       from MDAnalysis.topology.guessers import guess_angles
       import numpy as np

In [2]:u = mda.Universe('example.pdb', 'example.trr')

In [3]:angles_list = Angles(guess_angles(u.bonds), guessed=True)
       u.add_TopologyAttr(angles_list)

In [4]:coords = np.array([[ 83.37363434,  65.01402283,  35.03564835],
                          [ 82.28535461,  59.99816513,  34.94399261],
                          [ 81.19387817,  54.97631073,  34.85218048]],
                          dtype=np.float32)

In [5]:u.angles[0].atoms.positions = coords
       u.angles[0].atoms.positions
Out[5]:array([[ 83.37363434,  65.01402283,  35.03564835],
              [ 82.28535461,  59.99816513,  34.94399261],
              [ 81.19387817,  54.97631073,  34.85218048]], dtype=float32)

In [6]:u.angles[0].angle(), u.angles.angles()[0], u.angles._anglesSlow()[0]
Out[6]:(nan, 3.1412321525664901, -3.1415926535897931)

In [7]:#Angle.angle()
       #def angle(self):
       #    a = self[0].position - self[1].position
       #    b = self[2].position - self[1].position
       #    return np.rad2deg(np.arccos(np.dot(a, b) / (norm(a) * norm(b))))

In [8]:from MDAnalysis.lib.mdamath import norm
       a = coords[0] - coords[1]
       b = coords[2] - coords[1]
       x = np.dot(a, b) / (norm(a) * norm(b))
       x
Out[8]:-1.0000001

In [9]:np.arccos(x)
Out[9]:nan

```

### Currently version of MDAnalysis:
0.16.2

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1556

Changes made in this Pull Request:
 - Angle.angle now uses `mdamath.angle`

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/core/topologyobjects.py [hunk 1]
```diff
@@ -239,8 +238,7 @@ def angle(self):
         """
```
Comment (mnmelo) [inline package/MDAnalysis/core/topologyobjects.py]: Is the note in the docstring still applicable regarding the low precision? And if not, would it warrant making the corresponding tests stricter in that regard?

**Comment by mnmelo** [comment]
Just update that docstring to mention the higher precision and we're good to go.

**Comment by richardjgowers** [comment]
@mnmelo yeah I'm just updating the tests a little bit for pytest, give me a few mins :)

**Comment by richardjgowers** [comment]
ok should be good to go now

**Comment by mnmelo** [comment]
Sorry, didn't mean to be pushy. Just thought you had finalized it at that commit.

And thanks for the quick fix!

**Comment by richardjgowers** [comment]
@mnmelo yeah no worries, thanks for reviewing

**Comment by richardjgowers** [comment]
https://travis-ci.org/MDAnalysis/mdanalysis/jobs/257379205#L2769

Looks like we were right about the expected precision

**Comment by mnmelo** [comment]
Indeed. The mdamath version doesn't fare much better than what we were already using (but it still solves the precision-related 'nan' values).

Just roll back the test precision and the `angle()` docstring and it should be good.

**Pull request merged**